src/vsr/replica.zig

```diff
@@ -6763,7 +6763,12 @@ pub fn ReplicaType(
             assert(header.op < self.op or
                 self.journal.header_with_op(self.op).?.checksum == header.checksum);
 
-            if (!self.repair_header_would_connect_hash_chain(header)) {
+            if (self.journal.header_with_op(self.op).?.view == header.view) {
+                // Fast path for cases where the header being replaced is from the same view
+                // as the head. In this case, we can skip checking if our hash chain connects
+                // up till the head, as the primary for that view would have already done so in
+                // `on_prepare` (by invoking `panic_if_hash_chain_would_break_in_the_same_view`).
```

Comment:
Oh, this is even better than what I thought to do here! My idea was to check that `header.view == replica.view`, but comparing it with `header_head.view` is strictly better!

🥳